lib/beaker-pe/install/pe_utils.rb

```diff
@@ -448,6 +448,41 @@ def deploy_frictionless_to_master(host)
           end
         end
 
+        # Check for availability of required network resources
+        # @param [Array<Host>] hosts
+        # @param [Array<String>] network_resources, defaults to the value of options[:net_diag_hosts]
+        #                        if that option is not set we'll just verify connectivity to other test
+        #                        hosts.
+        #
+        # @example
+        #   verify_network_resources(hosts, network_resources)
+        #
+        # @return nil
+        #
+        # @api private
+        def verify_network_resources(hosts, network_resources = options[:net_diag_hosts])
```

Comment:
I recommend one change -- rather than pulling net_diag_hosts out of options in the defaults, and forcing this method to know about the options state, instead how about requiring network_resources, and let do_install() fish out options[:net_diag_hosts] and explicitly pass it. The do_install() method already has options locally.

Reply:
that makes sense